File: js/plugins/registries.js
/* =====================================================================
   REGISTRIES — one createRegistry() per extension point.
   Core code reads from these. Plugins write to them (through the SDK,
   never directly). Nothing here knows what a "Proxmox" or "Docker" is.
   ===================================================================== */
(function(HLM){
  "use strict";

  HLM.registries = {
    deviceTypes:          HLM.createRegistry("deviceTypes"),          // Step 5
    widgets:               HLM.createRegistry("widgets"),              // Step 4
    pages:                   HLM.createRegistry("pages"),                // nav pages/routes
    commands:                 HLM.createRegistry("commands"),
    dataProviders:              HLM.createRegistry("dataProviders"),      // Step 6
    notificationProviders:        HLM.createRegistry("notificationProviders"),
    settingsPanels:                 HLM.createRegistry("settingsPanels"),
    actions:                          HLM.createRegistry("actions"),        // toolbar/FAB/context-menu actions
    healthCalculators:                  HLM.createRegistry("healthCalculators"),
    alertRules:                           HLM.createRegistry("alertRules"),
    backgroundTasks:                        HLM.createRegistry("backgroundTasks"),
    themes:                                   HLM.createRegistry("themes"),
    translations:                               HLM.createRegistry("translations"),
    icons:                                        HLM.createRegistry("icons"),
  };
})(window.HLM = window.HLM || {});

File: backend/app/plugins/registries.py
"""
Instantiates every named registry — the backend mirror of
js/plugins/registries.js. Core code reads from these; plugins write to
them only through a PluginContext (see base.py), never directly.

Adding a new extension point later is a one-line addition here, same as
the frontend.
"""
from __future__ import annotations

from app.plugins.registry import Registry


class PluginRegistries:
    def __init__(self):
        self.collectors: Registry = Registry("collectors")
        self.commands: Registry = Registry("commands")
        self.notification_providers: Registry = Registry("notificationProviders")
        self.routers: Registry = Registry("routers")
        self.background_tasks: Registry = Registry("backgroundTasks")
        self.health_providers: Registry = Registry("healthProviders")


plugin_registries = PluginRegistries()
